engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/discovery-create.controller.js

```diff
@@ -1,92 +1,119 @@
+ * @requires translate
  *
  * @description
  *   Provides the functionality for the repo creation as part of
  *      repository discovery.
  */
-angular.module('Bastion.products').controller('DiscoveryFormController',
-    ['$scope', '$q', 'CurrentOrganization', 'Product', 'Repository', 'GPGKey', 'FormUtils',
-    function ($scope, $q, CurrentOrganization, Product, Repository, GPGKey, FormUtils) {
-
-        function productCreateError(response) {
-            $scope.working = false;
-            $scope.createRepoChoices.creating = false;
-            angular.forEach(response.data.errors, function (errors, field) {
-                if (angular.isUndefined($scope.productForm[field])) {
-                    $scope.productErrorMessages = 'An error occurred while saving the Product: ' + field + ' ' + errors;
-                } else {
-                    $scope.productForm[field].$setValidity('server', false);
-                    $scope.productForm[field].$error.messages = errors;
-                }
-            });
-        }
-
-        function repoCreateError(response) {
-            var currentlyCreating = $scope.currentlyCreating;
-            $scope.currentlyCreating = undefined;
-            $scope.createRepoChoices.creating = false;
-            $scope.productErrorMessages = response.data.errors;
-            currentlyCreating.form.$invalid = true;
-        }
+angular.module('Bastion.products').controller('DiscoveryCreateController',
+    ['$scope', '$q', 'CurrentOrganization', 'Product', 'Repository', 'GPGKey', 'FormUtils', 'translate',
+    function ($scope, $q, CurrentOrganization, Product, Repository, GPGKey, FormUtils, translate) {
+
+        $scope.errorMessages = [];
+        $scope.successMessages = [];
+
+        $scope.table = {
+            rows: $scope.$stateParams.rows,
+            resource: {
+                total: $scope.$stateParams.rows.length,
+                subtotal: $scope.$stateParams.rows.length
+            }
+        };
 
         function convertToResource(repo) {
-            return new Repository({
+            var repoParams = {
+                'docker_upstream_name': repo.dockerUpstreamName,
                 name: repo.name,
                 label: repo.label,
-                'content_type': 'yum',
+                'content_type': repo.contentType,
                 url: repo.url,
                 'product_id': $scope.createRepoChoices.existingProductId,
-                unprotected: $scope.createRepoChoices.unprotected
-            });
+                unprotected: $scope.createRepoChoices.unprotected,
+                'verify_ssl': $scope.createRepoChoices.verifySsl,
+                'upstream_username': $scope.createRepoChoices.upstreamUsername,
+                'upstream_password': $scope.createRepoChoices.upstreamPassword
+            };
+
+            if (repo.contentType === 'docker') {
+                repoParams.url = $scope.createRepoChoices.repositoryUrl;
+            } else {
```

Comment:
Do you need this else statement since `repoParams.url` is initialized to `repo.url` above?